### PR TITLE
Use float? for suggester fractional properties

### PR DESCRIPTION
**Pull request by russcam**
This commit updates TermSuggester to use float? instead of decimal? for
properties that take a float value.

Add XML documentation to suggesters

### CODE REVIEW

File: src/Nest/Mapping/Types/Specialized/Attachment/Attachment.cs [hunk 1]
```diff
@@ -5,6 +5,13 @@
 
 namespace Nest
 {
+	/// <summary>
```
Comment (Mpdreamz) [inline src/Nest/Mapping/Types/Specialized/Attachment/Attachment.cs]: `An indexed Attachement` feels like a SandCastle comment :cat: 

`<remarks>` should be used as extra documentation  about `why` something exists or a particular implementation inside the class was chosen. `<summary>` shows up in intellisense. Here I feel the remarks should be hoisted into the summary.

File: src/Nest/Search/Suggesters/CompletionSuggester/CompletionField.cs [hunk 2]
```diff
@@ -4,16 +4,29 @@
...
 	/// <summary>
-	/// Convenience class for use when indexing completion fields.
+	/// A completion field
 	/// </summary>
+	/// <remarks>
+	/// Convenience class for use when indexing completion fields.
+	/// </remarks>
 	public class CompletionField
 	{
+		/// <summary>
```
Comment (Mpdreamz) [inline src/Nest/Search/Suggesters/CompletionSuggester/CompletionField.cs]: Hate the hate on adding xmldocs but not fan of these style of SandCastle'esque summaries that feel generated. I rather have none instead. 

Further more to continue my nitpick for small summaries can we have a single line

```chsarp
//// <summary> The summary </summary>
```

File: src/Nest/Search/Suggesters/CompletionSuggester/CompletionField.cs [hunk 3]
```diff
@@ -4,16 +4,29 @@
 namespace Nest
 {
 	/// <summary>
-	/// Convenience class for use when indexing completion fields.
+	/// A completion field
 	/// </summary>
+	/// <remarks>
```
Comment (Mpdreamz) [inline src/Nest/Search/Suggesters/CompletionSuggester/CompletionField.cs]: Merge remarks and summary

File: src/Nest/Search/Suggesters/CompletionSuggester/FuzzySuggest.cs [hunk 4]
```diff
@@ -3,35 +3,61 @@
 
 namespace Nest
 {
+	/// <summary>
```
Comment (Mpdreamz) [inline src/Nest/Search/Suggesters/CompletionSuggester/FuzzySuggest.cs]: `FuzzySugester` is a bad name now that I am rereading it maybe `SuggesterFuzziness`?

Maybe link to https://www.elastic.co/guide/en/elasticsearch/reference/current/search-suggesters-completion.html#fuzzy ?

The comment itself as it is now adds little IMO.

Comment (russcam) [reply src/Nest/Search/Suggesters/CompletionSuggester/FuzzySuggest.cs]: ++ to `SuggesterFuzziness`; it better fits what it is.

I share some of the sentiment of GhostDoc/Sandcastle autogenerated XML comments. Do you think we should aim for consistency in applying `<summary>`?

File: src/Nest/Search/Suggesters/CompletionSuggester/CompletionField.cs (hunk 2 above)
Comment (russcam) [reply src/Nest/Search/Suggesters/CompletionSuggester/CompletionField.cs]: > for small summaries can we have a single line

I can do this. We'll also need to tweak our formatting configuration which always seems to format on multiple lines

File: src/Nest/Search/Suggesters/CompletionSuggester/FuzzySuggest.cs (hunk 4 above)
Comment (russcam) [reply src/Nest/Search/Suggesters/CompletionSuggester/FuzzySuggest.cs]: Looking at the naming convention of other types, `ISuggestFuzziness` might be more appropriate

**Comment by russcam** [comment]
I'm merging this in, to look to get a release out

**Pull request merged**